loki/transform/fortran_c_transform.py

```diff
@@ -477,17 +478,51 @@ def generate_c_kernel(self, routine):
         convert_to_lower_case(kernel)
 
         # Force pointer on reference-passed arguments
-        var_map = {}
+        to_be_dereferenced = []
```

Comment:
Readability of the method might benefit from putting the reference/dereference handling into a utility method maybe, and then simply calling `self.convert_args_to_references(kernel)`